text/0000-rand-crypto.md

```diff
@@ -0,0 +1,192 @@
+- Feature Name: rand-crypto
+- Start Date: 2017-08-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Background
+
+This RFC is an offshoot of the [Rand revision RFC], on the subject of:
+
+*   splitting rand into multiple crates
+*   splitting the current `Rng` trait into a crypto-oriented trait and
+    numerical-application-oriented trait
+*   impl of above trait(s) for OS-provided randomness
+*   naming conventions for above things
+
+# Summary
+[summary]: #summary
+
+Create a new crate, `crypto-rand`, with only the following contents:
+
+```rust
+pub enum Error {
+    pub Unspecified,
+}
+
+pub trait Generator {
+    fn try_fill(&self, dest: &mut [u8]) -> Result<(), Error>;
+}
```

Comment:
A bit of name bike-shedding:

Not sure if we need `try_fill` instead of simply `fill`. I don't think this method will conflict with `Rng` trait, as they'll be rarely used together.

Also I think `Generator` is a bit too long, how about `Csrng`? (you propose something similar in alternatives) It's a bit more cryptic, but will be in a good alignment with my further proposal regarding crate names. (I don't fully agree with Brian regarding the need to remove "r" from "rng" just because crate name is based on "random")